pkg/apis/smith/v1/types.go

```diff
@@ -194,11 +197,27 @@ type Resource struct {
 	Name ResourceName `json:"name"`
 
 	// Explicit dependencies.
-	DependsOn []ResourceName `json:"dependsOn,omitempty"`
+	References []Reference `json:"reference,omitempty"`
 
 	Spec ResourceSpec `json:"spec"`
 }
 
+// +k8s:deepcopy-gen=true
+// Refer to a part of another object
+type Reference struct {
+	Name     ReferenceName `json:"name,omitempty"`
+	Resource ResourceName  `json:"resource"`
```

Comment:
This should probably be `ResourceName ResourceName`. But then it is a bit too verbose. WDYT?